onnxruntime/core/optimizer/compute_optimizer/compute_optimizer.cc

```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-#ifdef ENABLE_TRAINING
+#ifdef ENABLE_TRAINING_CORE
 #include <onnx/defs/attr_proto_util.h>
```

Comment:
if the whole file is only for training build, could we just put it under the training folder, and only include it in training build in the cmake, instead of have the ifdef in the code?

Reply:
this is added by me, I put it here for easily enabling it for inferencing later.